src/components/structures/MatrixChat.js

```diff
@@ -304,6 +305,9 @@ module.exports = React.createClass({
                 });
                 break;
             case 'view_room':
+                // by default we autoPeek rooms, unless we were called explicitly with
+                // autoPeek=false by something like RoomDirectory who has already peeked
+                this.setState({ autoPeek : payload.auto_peek === false ? false : true });
```

Comment:
I don't understand why you are not just doing:

```javascript
this.setState({ autoPeek : payload.auto_peek });
```

As that is functionally identical to what you have written here. You're doing `===` so there will be no type coercion for `undefined`.